### DISCUSSION TITLE
if I change B in "this.WhenAnyValue(x=>x.A.B.C)" to a new object, will ReactiveUI dispose subscription of previous B object?

### DISCUSSION THREAD

**Discussion by LJN-hzleaper** [reactiveui/ReactiveUI#3709] [Q&A] [answered]
When using ReactiveUI's WhenAnyValue, if I change B in "this.WhenAnyValue(x=>x.A.B.C)" to a new object, will ReactiveUI dispose subscription of previous B object?

**Comment by glennawatson** [comment, marked as answer]
It will dispose of the subscription, providing the B property is a INotifyPropertyChanged or other mechanism that ReactiveUI knows about notifications from.
